<a href="https://colab.research.google.com/github/KhurramDevOps/Quarter-02/blob/master/04_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **aisuite**



In [1]:
!pip install aisuite

In [2]:
!pip install 'aisuite[anthropic]'

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 863.9/863.9 kB 11.1 MB/s eta 0:00:00


In [3]:
!pip install 'aisuite[all]'

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 103.5/103.5 kB 2.0 MB/s eta 0:00:00


In [8]:
import os

# Set the API keys
os.environ['GOOGLE_API_KEY'] = 'GEMINI_API_KEY'
os.environ['ANTHROPIC_API_KEY'] = 'Anthropic_API_KEY'


In [14]:
import os
import aisuite as ai

# Set the Gemini API Key
os.environ['GOOGLE_API_KEY'] = 'your-gemini-api-key'  # Replace with your actual key

# Initialize the client
client = ai.Client()

# Models you want to use
models = ["google:gemini-1.5-flash", "anthropic:claude-3-5-sonnet-20240620"]

# Define messages
messages = [
    {"role": "system", "content": "Respond in Pirate English."},
    {"role": "user", "content": "Tell me a joke."},
]

# Iterate over models and get responses
for model in models:
    response = client.chat.completions.create(
        model=model,
        messages=messages,
        temperature=0.75
    )
    print(response.choices[0].message.content)


OSError: Missing one or more required Google environment variables: GOOGLE_PROJECT_ID, GOOGLE_REGION, GOOGLE_APPLICATION_CREDENTIALS. Please refer to the setup guide: /guides/google.md.

### After this error , i know that i need the Google_project_ID, Google_Rejion, Google_Application_Credentials in json format . In next steps i try my best to figure it out!

In [21]:
from google.cloud import aiplatform

# Set the region
aiplatform.init(
    project="my-gemini-api-444713",
    location="asia-south1"
)


In [22]:
!gcloud config set compute/region asia-south1
!gcloud config set compute/zone asia-south1-a


ERROR: (gcloud.config.set) You do not currently have an active account selected.
Please run:

  $ gcloud auth login

to obtain new credentials.

If you have already logged in with a different account, run:

  $ gcloud config set account ACCOUNT

to select an already authenticated account to use.
ERROR: (gcloud.config.set) You do not currently have an active account selected.
Please run:

  $ gcloud auth login

to obtain new credentials.

If you have already logged in with a different account, run:

  $ gcloud config set account ACCOUNT

to select an already authenticated account to use.


In [23]:
import os

# Set Google environment variables (replace with actual values)
os.environ['GOOGLE_PROJECT_ID'] = 'my-gemini-api-444713'
os.environ['GOOGLE_REGION'] = 'asia-south1'  # e.g., 'us-central1'
os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = '/content/my-gemini-api-444713-fac594037688.json'


In [24]:
print(os.getenv('GOOGLE_PROJECT_ID'))
print(os.getenv('GOOGLE_REGION'))
print(os.getenv('GOOGLE_APPLICATION_CREDENTIALS'))


my-gemini-api-444713
asia-south1
/content/my-gemini-api-444713-fac594037688.json


### So, in above code i get the project id from google claud where a i created the project some days ago , after that after alot of struggle and i finally get the "GOOGLE_APPLICATION_CREDENTIALS in JSon format" and i downloaded it and add the path .


### Then i need Google_Rejion and after couple of hours i didn't find it ! then i say gpt to tell me make some guess to try and i try some of these but the issue is same coming but still i use the "asia-south1". then i try below!




In [27]:
import os
import aisuite as ai

# Set the Google Cloud environment variables
os.environ['GOOGLE_PROJECT_ID'] = 'my-gemini-api-444713'  # Replace with your Google Cloud project ID
os.environ['GOOGLE_REGION'] = 'asia-south1'  # e.g., 'us-central1'
os.environ['GOOGLE_APPLICATION_CREDENTIALS'] = '/content/my-gemini-api-444713-fac594037688.json'  # Replace with path to your service account JSON file

# Set the Gemini API Key
os.environ['GOOGLE_API_KEY'] = 'your-gemini-api-key'  # Replace with your actual key

# Initialize the client
client = ai.Client()

# Models you want to use
models = ["google:gemini-1.5-flash", "anthropic:claude-3-5-sonnet-20240620"]

# Define messages
messages = [
    {"role": "system", "content": "Respond in Pirate English."},
    {"role": "user", "content": "Tell me a joke."},
]

# Iterate over models and get responses
for model in models:
    response = client.chat.completions.create(
        model=model,
        messages=messages,
        temperature=0.75
    )
    print(response.choices[0].message.content)


PermissionDenied: 403 Vertex AI API has not been used in project my-gemini-api-444713 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/aiplatform.googleapis.com/overview?project=my-gemini-api-444713 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry. [links {
  description: "Google developers console API activation"
  url: "https://console.developers.google.com/apis/api/aiplatform.googleapis.com/overview?project=my-gemini-api-444713"
}
, reason: "SERVICE_DISABLED"
domain: "googleapis.com"
metadata {
  key: "service"
  value: "aiplatform.googleapis.com"
}
metadata {
  key: "consumer"
  value: "projects/my-gemini-api-444713"
}
]

### ***My Final Conclusion :*** After alot of tries and search i dont solve this . Because, i think there's issues in Google Rejion Probably, and also there a issue of "403 Vertex Api error" that you see above . So, i go to Google cluad and try to enable that and it want billing to open the Api and enables it to use . i think after billing of vertex api this will Probably works .
  ### At the end i find these main issues in my opinion May be there's some more Error but im stuck in these errors. Otherwise we have to try other free API's may be they can work or not. I try my best to do!